python/openassetio_traitgen/templates/cpp/trait.hpp.in

```diff
@@ -109,19 +110,19 @@ public:
   }
 
   /**
-   * Gets the value of the {{ property.id }} property or throws if
-   * not found or is of an unexpected type.
+   * Gets the value of the {{ property.id }} property. Returns an empty
+   * optional if not found or is of an unexpected type.
    *
    * {{ property.description |  wordwrap(69, wrapstring="\n* ") | indent(4) }}
    */
-  [[nodiscard]] {{ VarType }} get{{ VarMethodName }}() const {
-    if (property::Value value; traitsData_->getTraitProperty(&value, kId, "{{ property.id }}")) {
+  [[nodiscard]] std::optional<{{ VarType }}> get{{ VarMethodName }}() const {
+  if (property::Value value; traitsData_->getTraitProperty(&value, kId, "{{ property.id }}")) {
```

Comment:
[fix] Lost some indentation here that makes the generated output a bit weird.

Reply:
https://github.com/OpenAssetIO/OpenAssetIO-TraitGen/pull/63/commits/03b3c1df60455e678dbe8a454e0b2c467abe10f4